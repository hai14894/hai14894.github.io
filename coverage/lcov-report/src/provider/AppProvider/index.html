
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/provider/AppProvider</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/provider/AppProvider</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">48.05% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>37/77</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>3/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">37.93% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>11/29</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>34/68</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="App.actions.js"><a href="App.actions.js.html">App.actions.js</a></td>
	<td data-value="45.45" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 45%"></div><div class="cover-empty" style="width: 55%"></div></div>
	</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="22" class="abs low">10/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	</tr>

<tr>
	<td class="file high" data-value="App.provider.js"><a href="App.provider.js.html">App.provider.js</a></td>
	<td data-value="88" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88" class="pct high">88%</td>
	<td data-value="25" class="abs high">22/25</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="24" class="abs high">21/24</td>
	</tr>

<tr>
	<td class="file low" data-value="App.reducer.js"><a href="App.reducer.js.html">App.reducer.js</a></td>
	<td data-value="16.67" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 16%"></div><div class="cover-empty" style="width: 84%"></div></div>
	</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="30" class="abs low">5/30</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="30" class="abs low">5/30</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Jul 22 2021 14:23:53 GMT+1000 (Australian Eastern Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    