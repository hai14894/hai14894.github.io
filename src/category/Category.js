import React, { useEffect,useState } from "react";
import axios from "axios";
import HeroSection from "../components/HeroSection/HeroSection.js";
import NavBar from "../components/NavBar/NavBar.js";
import Item from "./Item"
import "./Category.css";

const Category = () => {
    const[items,setItems] = useState(null)
  const getData = async function () {
    await axios
      .get("products.json", {
        headers: {
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      })
      .then((response) => {
        setItems(response.data)
      });
  };
  useEffect(() => {
    getData();
  }, []);
 
  console.log(items);
 
  return (
    <div className="Category">
      <NavBar />
      <HeroSection />
      <div>
        {items &&
          items.map((item, key) => {
            return (
              <div key={key}>
                <Item
                  title={item.title}
                  brand={item.brand}
                  price={item.price}
                />
              </div>
            );
          })}
      </div>

    </div>
  );
};

export default Category;
