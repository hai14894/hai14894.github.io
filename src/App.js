import React, { Component } from 'react';
import { Link, Route } from 'react-router-dom';
import logo from './logo.svg';
import './App.css';
import Category from './category/Category';
import Cart from './cart/Cart';
import Product from './product/Product';

const App = () => {
    return (
        <div className="App">
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <h1 className="App-title">Storefront Assignment</h1>
            </header>
            <p className="App-intro">
                To get started, delete this header and introduction, and begin
                building your app in the provided components.
            </p>
            <p className="App-intro">
                We've setup the bare minimum you need to get started, but feel
                free to add as many components as you see fit.
            </p>

            <header>
                <Link to="/cart">My Cart</Link>
            </header>

            <Route exact path="/" component={Category} />
            <Route path="/cart" component={Cart} />
            <Route path="/product/:id" component={Product} />
        </div>
    );
};

export default App;
