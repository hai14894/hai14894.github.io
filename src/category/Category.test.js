import React from "react";
import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { AppContext } from "../provider/AppProvider/App.provider";
import Category from "./Category";

const MockProduct = [
  {
    title: "Blue Stripe Stoneware Plate",
    brand: "Kiriko",
    price: 40,
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    image: "blue-stripe-stoneware-plate.jpg",
  },
  {
    title: "Hand Painted Blue Flat Dish",
    brand: "Kiriko",
    price: 28,
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    image: "hand-painted-blue-flat-dish.jpg",
  },
];

const mockProviderValue = {
  state: {
    loading: false,
    products: MockProduct,
    cart: [
      {
        brand: "Kiriko",
        count: 1,
        description:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
        id: 2,
        image: "hand-painted-blue-flat-dish.jpg",
        price: 28,
        title: "Hand Painted Blue Flat Dish",
      },
    ],
  },
};

describe("Category Page test suit", () => {
  const addToCart = jest.fn();
  const removeFromCart = jest.fn();

  it("Should render correct product from Provider", () => {
    const { getByText } = render(
      <AppContext.Provider
        value={{ ...mockProviderValue, actions: { addToCart, removeFromCart } }}
      >
        <Router>
          <Category />
        </Router>
      </AppContext.Provider>
    );

    expect(getByText("Blue Stripe Stoneware Plate")).toBeInTheDocument();
    expect(getByText("Hand Painted Blue Flat Dish")).toBeInTheDocument();
  });

  it('Should render "View Details" with correct link to product page', () => {
    const { container } = render(
      <AppContext.Provider
        value={{ ...mockProviderValue, actions: { addToCart, removeFromCart } }}
      >
        <Router>
          <Category />
        </Router>
      </AppContext.Provider>
    );

    fireEvent.mouseOver(container.querySelector('[data-testid="category1"]'));
    expect(
      container.querySelector('[data-testid="category1"] a')
    ).toHaveAttribute("href", "/product/1");
  });

  it('Should fire action AddToCart when click "Add To Cart" button on overlay', () => {
    const addToCart = jest.fn();
    const { container } = render(
      <AppContext.Provider
        value={{ ...mockProviderValue, actions: { addToCart, removeFromCart } }}
      >
        <Router>
          <Category />
        </Router>
      </AppContext.Provider>
    );
    fireEvent.mouseOver(container.querySelector('[data-testid="category1"]'));

    fireEvent.click(
      container.querySelector('[data-testid="category1"] button')
    );

    expect(addToCart).toHaveBeenCalledWith({
      ...MockProduct[0],
      id: 1,
      count: 1,
    });
  });

  it('Should fire action RemoveCart with correct id when click "X" button on CartPopup', () => {
    const removeFromCart = jest.fn();
    const { container } = render(
      <AppContext.Provider
        value={{ ...mockProviderValue, actions: { addToCart, removeFromCart } }}
      >
        <Router>
          <Category />
        </Router>
      </AppContext.Provider>
    );

    fireEvent.click(container.querySelector("#nav-cart"));
    fireEvent.click(container.querySelector('[data-testid="removeFromCart0"]'));

    expect(removeFromCart).toHaveBeenCalledWith(2);
  });

  it("Should render with correct snapshot", () => {
    const { container } = render(
      <AppContext.Provider
        value={{ ...mockProviderValue, actions: { addToCart, removeFromCart } }}
      >
        <Router>
          <Category />
        </Router>
      </AppContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
