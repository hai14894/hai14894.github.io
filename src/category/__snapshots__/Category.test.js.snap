// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Page test suit Should render with correct snapshot 1`] = `
<div>
  <div>
    <div
      class="css-ybrkw5"
    >
      <nav
        class="css-1ikv20o navbar navbar-expand-md navbar-light sticky-top"
      >
        <a
          class="navbar-brand"
          href="/"
          id="logo-branch"
        >
          <img
            alt="logo"
            src="/media/logo.png"
          />
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="inner-nav navbar-nav"
          >
            <a
              class="nav-link"
              data-rb-event-key="/"
              href="/"
            >
              HOME
            </a>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
                role="button"
              >
                SHOP
              </a>
            </div>
            <a
              class="nav-link"
              data-rb-event-key="/"
              href="/"
            >
              JOURNAL
            </a>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
                role="button"
              >
                MORE
              </a>
            </div>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
                id="nav-cart"
                role="button"
              >
                MY CART (1)
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div
      class="css-1692lr3"
    >
      <div
        class="css-54yhhp"
      >
        <div
          class="css-82aryr"
        >
          <div
            class="css-a6lbtd"
          >
            Plates
            <hr
              style="color: white; margin: 0px;"
            />
          </div>
          <div
            class="css-dgitsn"
          >
            Lorrem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerir auctor a eget arcu. Curabitur ac pharetra nisl sit
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-x0sfrr"
    >
      <div
        class="css-1i4ma4x"
      >
        <div
          class="css-1iyoj2o"
          data-testid="category1"
        >
          <div
            class="css-1kirz7d"
          >
            <div
              class="css-g2nr4m"
            />
            <div
              class="css-6gzdt7"
              width="100%"
            />
          </div>
          <div
            class="css-jyu1kt"
          >
            <div
              class="css-qnq5vm"
            >
              Kiriko
            </div>
            <div
              class="css-1bysjuy"
            >
              Blue Stripe Stoneware Plate
            </div>
            <div
              class="css-1no1bop"
            >
              $40.00
            </div>
          </div>
        </div>
        <div
          class="css-1iyoj2o"
          data-testid="category2"
        >
          <div
            class="css-1kirz7d"
          >
            <div
              class="css-g2nr4m"
            />
            <div
              class="css-1fi6bgp"
              width="100%"
            />
          </div>
          <div
            class="css-jyu1kt"
          >
            <div
              class="css-qnq5vm"
            >
              Kiriko
            </div>
            <div
              class="css-1bysjuy"
            >
              Hand Painted Blue Flat Dish
            </div>
            <div
              class="css-1no1bop"
            >
              $28.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
